### DISCUSSION TITLE
[4.0]  IMPROVEMENT categories

### DISCUSSION THREAD

**Discussion by Open2logic** [joomla/joomla-cms#47183] [2. Cold Features]
A website can have corporate information, for example:
about us?, address, services, tec.

But you can also have organized content: classified ads,, directories, news, etc.
EXAMPLE:
Categoriy parent named: SPORTS
Categories children named: FOOBALL, BASKET, GOLF, TENNIS, etc

Now there are two deficiencies

**FIRST IMPROVEMENT: ADD ANOTHER CATEGORY BUTTOM**
Joomla can not show articles of several chosen categories

**NOW 1 category**
![joomla4-1](https://user-images.githubusercontent.com/9007607/61420045-168e5980-a901-11e9-849e-5e0109b5b0c3.JPG)

**and show EDIT or CLEAR**
![joomla4-2](https://user-images.githubusercontent.com/9007607/61420081-43427100-a901-11e9-9c6c-3d09a0b70a2f.JPG)

**THIS IS BETTER, button for ADD ANOTHER CATEGORY**
![joomla4-another](https://user-images.githubusercontent.com/9007607/61420241-e2ffff00-a901-11e9-99b6-064819e1aec3.jpg)

**SECOND IMPROVEMENT: CHOSEN CATEGORIES IN FORM**
Now a user who can add an article necessarily sees all the categories of a parent or a predetermined category

**NOW all category or default category**
![cat-joomla4](https://user-images.githubusercontent.com/9007607/61420539-476f8e00-a903-11e9-934f-bf56ca275069.JPG)

**THIS IS BETTER, select ctegories to show for the user in the form**
![user-4](https://user-images.githubusercontent.com/9007607/61420755-4b4fe000-a904-11e9-95bb-50ce20d905de.jpg)

They usually put a predetermined category, In this example SPORTS, but they can not choose the children catergoies FOOBALL, BASKET, GOLF, TENNIS, etc.

THANKS

**Comment by HLeithner** [comment]
You should use tags for this. 
Joomla categories are designed to only have one category per item. This generates also an unique url for the article.

**Comment by micker** [comment]
ar use a CCK (some of this have add multicategory)

**Comment by ghost** [comment]
closed for Comments above. Please reopen if further Discussion needed.

**Comment by Open2logic** [comment]
My English is very bad, but really three people nobody has understood what he has exposed.

**Comment by Open2logic** [comment]
I do not speak of articles in several categories, I do not speak of a cck, I do not speak of labels, I speak of being able to choose to show several categories in blog format, something very necessary in directories, advertisements, etc.

**Comment by Open2logic** [comment]
> You should use tags for this.
> Joomla categories are designed to only have one category per item. This generates also an unique url for the article.

When you associate an item to a parent category, you are showing articles of various categories., 
It's about being able to choose which specific categories.

**Comment by brianteeman** [comment]
If you have a category structure like this
Blog
-- news
-- joomla
-- web

And then create a blog menu item to point to the category BLOG and include subcategories then you achieve what you are looking for. That is exactly how I have created my home page (https://brian.teeman.net). 
![image](https://user-images.githubusercontent.com/1296369/61445854-119fc900-a946-11e9-963d-dc647c8fab36.png)

**Comment by Open2logic** [comment]
Bravo Brian, thank you very much, it's exactly that, but it's not very intuitive, nor did I stop to understand what the explanatory text says, how many users do not know this functionality for blogs?

**Comment by Open2logic** [comment]
Brian certainly what I indicated to choose certain categories was a request that you liked in the past but was dismissed because Joomla 3 opens a modal window to choose the category and it looked complicated, that's why I wanted to recover the request for Joomla 4 but with a better UX than the current one, the current one and thinking carefully about the current function, there are still limitations that depend on the order and number chosen in the subcategories to be displayed.

**Comment by brianteeman** [comment]
I dont think making the user select each category instead of simply checking to include subcategories is any easier

**Comment by Open2logic** [comment]
But currently the subcategories must belong to the same father and also have to be in order, are limitations for a blog

**Comment by Open2logic** [comment]
https://github.com/joomla/joomla-cms/issues/15249#issuecomment-302823905

**Comment by brianteeman** [comment]
You see a limitation - I see good data structure

Not sure why you post that link as it was related to submitting content on the front end and not to displaying content

**Comment by Open2logic** [comment]
Show the link because it also has a reference to the subcategories added in the menu item not in the blog view but is in the view of this form to add article. for example Brian on your website: https: //brian.teeman.net).
If you create a form to add an article associated with your parent category, what categories and how many will the user see?
I'm going to try it as you indicate.

**Comment by Open2logic** [comment]
Fo rme not work, include a number in subcategories, the intem to blog show all articles of all categories.

![multi](https://user-images.githubusercontent.com/9007607/61451562-ff2f8a80-a959-11e9-8343-0881aa80d770.JPG)


I will try a new instance

**Comment by Pablo234** [comment]
I agree with this improvement.

Also include the possibility to create a new article just in selected category and subcategories.


For example:

You have a news blog. You want people write news, but you have 2 subactegories (sports, health).


Now joomla just allow you to select category (news) or subcategory (sports or health). The new must be in subcategory, not in parent.

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25619">issues.joomla.org/tracker/joomla-cms/25619</a>.</sub>

**Comment by Open2logic** [comment]
I am a Joomla course teacher (7 years) and when I put the exercise of creating a Blog all students (who have a basic vision of the needs) ask me the same question: why can't I choose the subcategories). This improvement is simple but necessary for Blog, ads websites, news websites, directories, etc. Joomla would be perfect for this type of websites with this improvement.
Thanks for seeing it.

**Comment by Open2logic** [comment]
> If you have a category structure like this
> Blog
> -- news
> -- joomla
> -- web
> 
> And then create a blog menu item to point to the category BLOG and include subcategories then you achieve what you are looking for. That is exactly how I have created my home page (https://brian.teeman.net).
> ![image](https://user-images.githubusercontent.com/1296369/61445854-119fc900-a946-11e9-963d-dc647c8fab36.png)


I tried this

Categories and subcategories of a news website::
![categories](https://user-images.githubusercontent.com/9007607/62270464-5d6a6c00-b436-11e9-8c36-943343fb4c78.JPG)

And its corresponding articles
![articles](https://user-images.githubusercontent.com/9007607/62270866-60199100-b437-11e9-87e1-6cb80cc27399.JPG)

I create a blog menu button
![boton](https://user-images.githubusercontent.com/9007607/62270956-a242d280-b437-11e9-82e5-aedca48388cd.JPG)

And the blog layout options
Include 2 subcategories (they should be the closest to the father) Football & Bastketball
![options](https://user-images.githubusercontent.com/9007607/62271122-1b422a00-b438-11e9-9677-f5ae3540eb69.JPG)

But the website show this:
![view1](https://user-images.githubusercontent.com/9007607/62271561-33667900-b439-11e9-98c7-1e39c194947f.JPG)
![new2](https://user-images.githubusercontent.com/9007607/62271570-38c3c380-b439-11e9-95b1-26d4c59928cc.JPG)

NO WORK, the website show  Articles of 4 categories no of 2.
And where are the number of columns and column order options?

Joomla! 4.0.0-alpha10

**Comment by SharkyKZ** [comment]
`Include Subcategories` option sets subcategory level limit, not the total number of included subcategories.

**Comment by Open2logic** [comment]
Sorry this is the option, in Category Tab.
![levl](https://user-images.githubusercontent.com/9007607/62290136-e4343e80-b460-11e9-843e-bb2125a6c1a9.JPG)

But if it is true that the text indicates LEVELS of subcategories to show no number of subcategories to display.
But when I tell you to show me categories of the second level it also shows those of the first level (which seems logical)
So it is impossible to display articles from several selected categories, I think this new functionality is very necessary.

thanks
